### DISCUSSION TITLE
Is usage of a subquery in `selectinload` `extra_criteria` officially supported?

### DISCUSSION THREAD

**Discussion by sweetro** [sqlalchemy/sqlalchemy#7486] [Usage Questions]
Hey everyone, raising this here before filing an issue.

Is using a subquery within a `selectinload` supported? For example

```
subquery = some_query.subquery()

session.query(A).options(selectinload(A.bs.and_(B.some_id == subquery.c.some_id)).populate_existing()
```

If it is supported, I'm seeing the query return correct results initially, and then intermittent incorrect results once the conditions of the subquery change, suggesting to me there may be some caching issue on the subquery similar to https://github.com/sqlalchemy/sqlalchemy/issues/6881#issuecomment-898457025. Will file a more thorough issue if this _should_ be supported.

**Comment by zzzeek** [comment]
it's absolutely supported so please create a succinct test case.

**Comment by sweetro** [comment]
Thanks @zzzeek! following the minimal you wrote from #6881, I created https://github.com/sqlalchemy/sqlalchemy/issues/7489
